### PR TITLE
[ENB-7566] Change domain names for stage, remove personalization logic for not signed in

### PR DESCRIPTION
**Pull request by swamu**
Bug 1:
prod: https://www.adobe.com/experienceedge/ee/v2/interact (based on domain name)
stage: https://www.stage.adobe.com/experienceedge/ee/v2/interact
dev/hlx: https://edge.adobedc.net/ee/v2/interact

Bug 2:
Fix campaign applying without target=on ( when personalization is present)

https://main--cc--adobecom.hlx.live/drafts/suhjain/swati-pages/photoshop-v2?target=on&milolibs=ENB-7556-update-api-endpoints--milo--swamu#

Resolves: [ENB-7556](https://jira.corp.adobe.com/browse/ENB-7556)

Test URLs:
Before: https://main--milo--adobecom.aem.page/?martech=off
After: https://ENB-7556-update-api-endpoints--milo--swamu.aem.page/?martech=off

### CODE REVIEW

File: libs/martech/helpers.js [hunk 1]
```diff
@@ -338,6 +338,30 @@ function updateAMCVCookie(ECID) {
...
+  const { host } = window.location;
+  const query = PAGE_URL.searchParams.get('env');
+  const url = 'https://edge.adobedc.net/ee/v2/interact';
+  const SLD = PAGE_URL.hostname.includes('.aem.') ? 'aem' : 'hlx';
+
+  /* c8 ignore start */
+  if (query || host.includes('localhost') || host.includes(`${SLD}.page`)
+    || host.includes(`${SLD}.live`)) {
+    return url;
+  }
```
Comment (mokimo) [inline libs/martech/helpers.js:352]: You can also do just check for `.live` or `.page` and get rid of the SLD part

Comment (swamu) [reply libs/martech/helpers.js]: Done @mokimo

**Review by amitbikram — APPROVED** [review]

**Comment by Nadeemakramn** [comment]
Changes Approved.

File: libs/martech/helpers.js [hunk 2]
```diff
@@ -330,12 +326,36 @@ function extractECID(data) {
...
+
+  /* c8 ignore start */
+  if (host.includes('stage.adobe')
+    || host.includes('corp.adobe')
+    || host.includes('graybox.adobe')) {
+    return 'https://www.stage.adobe.com/experienceedge/ee/v2/interact';
+  }
+
+  const { origin } = window.location;
+  return `${origin}/experienceedge/ee/v2/interact`;
```
[the remark this answers is not in the document]
Comment (swamu) [reply libs/martech/helpers.js:358]: @mokimo considered that method, had changes in the env, different links for stage and live/page hence did this.

**Pull request merged**